src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -445,6 +450,22 @@ export function useAnimatedStyle<T extends AnimatedStyle>(
 
   dependencies.push(sharableViewDescriptors);
 
+  const areListsEqual = (a1: DependencyList, a2: DependencyList) => {
+    for (let i = 0; i < (a1?.length || 0); ++i) {
+      if (a1?.[i] !== a2?.[i]) return false;
+    }
+    return true;
+  };
```

Comment:
Can we define this function in top-level and also use `function areListsEqual(...) {}` syntax?